vm/actor/src/builtin/market/mod.rs

```diff
@@ -665,35 +665,38 @@ impl Actor {
         let proposals = DealArray::load(&st.proposals, rt.store()).map_err(|e| {
             e.downcast_default(ExitCode::ErrIllegalState, "failed to load deal proposals")
         })?;
-
-        let mut pieces: Vec<PieceInfo> = Vec::with_capacity(params.deal_ids.len());
-        for deal_id in params.deal_ids {
-            let deal = proposals
-                .get(deal_id as usize)
+        let mut commds = Vec::with_capacity(params.inputs.len());
+        for comm_input in params.inputs.iter() {
```

Comment:
style: It'd be nice to do this with a `for_each` and `map`/`collect` combinators if possible

Reply:
Those are usually best for smaller portions of logic, imo. Closures complicate things when handling references to multiple variables in larger sections of code.

Reply:
Error handling gets a little tricky as well. I dont think this should be changed to use combinators.

Reply:
Excuses ... https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.try_for_each

Combinators are easier to read because the combinator methods describe the transform/aggregation/filter. As for references, in my experience running into roadblocks when passing references  into closures is the exception not the common case; especially in a small function like this.

It's just a style recommendation though, feel free to resolve if you aren't enticed.

Reply:
These aren't just "excuses", Jorge. You seem to be implying we're excusing mediocrity, and that simply isn't the case. I don't disagree they make some smaller sections of code more readable, but unless we're going full ham on FP with transducers / reducer composition, there are fundamental limitations and differences to how chained higher-order functions behave, and they're not always the right tool for the job.

One can't be prescriptive when building such things, especially with a systems programming language built around ownership types. This isn't JavaScript or Haskell. Ideally we lean into what's idiomatic for the language.

Reply:
Excuses I tell you!

Reply:
:joy_cat: